src/dataset-builder/DatasetBuilderDetails.ts

```diff
@@ -72,14 +73,24 @@ export const DatasetBuilderDetails = ({ datasetId }: DatasetBuilderDetailsProps)
     datasetRoles.status === 'Ready' ? _.intersection(['admin'], datasetRoles.state).length > 0 : false;
 
   useOnMount(() => {
-    void loadDatasetDetails(() => DataRepo().dataset(datasetId).details());
-    void loadDatasetRoles(() => DataRepo().dataset(datasetId).roles());
+    void loadDatasetDetails(
+      withErrorReporting(`Error loading dataset details for dataset ${datasetId}`)(() =>
+        DataRepo().dataset(datasetId).details()
```

Comment:
While looking at this I noticed similar error handling code in `useImportRequest.ts`:
```
  let snapshot: Snapshot;
  try {
    snapshot = await DataRepo().snapshot(snapshotId).details();
  } catch (err: unknown) {
    throw new Error('Unable to load snapshot.');
  }
```

I assume that using `withErrorReporting()` is better UX as it reports the error, but I'm wondering if there are different philosophies about error handling in terra-ui.

Reply:
`withErrorReporting` is better since it reports the error, yeah. It is generally the preferred approach.

Reply:
The intent of that snippet from `useImportRequest.ts` is to replace the original error from the backend with a more general error message. The error there is still shown to the user, but it's shown on the page instead of in an error notification.

Generally, I think the notification approach is best for errors in update operations, etc. If an error occurs loading data, you may want to show a notification, but you probably also want some persistent status message wherever the loaded content was going to be displayed. Otherwise, after the notification expires, there's just an empty panel with no indication of what went wrong.